res/controllers/Icon-iControls-scripts.js

```diff
@@ -0,0 +1,187 @@
+                    }
+                },
+            });
+        }
+    }
+
+    class VelocityPot extends components.Pot {
+        constructor(params) {
+            super(Object.assign({
+                softTakeover: false,
+                relative: true,
+            }, params));
+        }
+        inValueScale(value) {
+            if (value & 0x40) {
+                return this.inGetParameter() - ((value & ~0x40) / 100);
+            } else {
+                return this.inGetParameter() + (value / 100);
+            }
+        }
+    }
+
+    class Controller extends components.ComponentContainer {
+        constructor() {
+            super({});
+            this.activeDeck = new Deck();
+
+            this.faders = new Array(9);
+            this.knobs = new Array(9);
+            this.pflButton = new Array(4);
+            this.nextEffectButton = new Array(4);
+            this.slipButton = new Array(4);
+            this.superButton = new Array(4);
+            for (let i = 0; i < 4; i++) {
+                const channel = mapIndexToChannel(i);
+                const group = `[Channel${channel}]`;
+                this.faders[i] = new components.Pot({
+                    group: group,
+                    midi: [0xB0+i, 0x0D],
+                    inKey: "volume",
+                    softTakeover: true,
+                });
+                this.faders[i+4] = new components.Pot({
+                    group: group,
+                    midi: [0xB0+i, 0x0D],
+                    inKey: "rate",
+                    softTakeover: true,
+                });
+                this.knobs[i] = new VelocityPot({
+                    group: group,
+                    midi: [0xB0, 0x10+i],
+                    inKey: "pregain",
+                });
+                this.knobs[i+4] = new VelocityPot({
+                    group: `[QuickEffectRack1_${group}]`,
+                    midi: [0xB0, 0x14+i],
+                    inKey: "super1",
+                });
+                this.pflButton[i] = new components.Button({
+                    group: group,
+                    midi: [0x90, 0x08+i],
+                    key: "pfl",
+                    type: components.Button.prototype.types.toggle,
+                });
+                this.slipButton[i] = new components.Button({
+                    group: group,
+                    midi: [0x90, 0x10+i],
+                    key: "slip_enabled",
+                    type: components.Button.prototype.types.toggle,
+                });
+                this.nextEffectButton[i] = new components.Button({
+                    group: `[QuickEffectRack1_${group}]`,
+                    midi: [0x90, 0x0C+i],
+                    // The manual says this is deprecated, but the replacement
+                    // it gives is for something else, not the quick effects
+                    // rack. Unclear to me what should be done here.
+                    key: "next_chain",
```

Comment:
If you follow the link in the library it brings you to `next_chain_preset` which is also implemented for the `[QuickEffectRack1_*]` group

![image](https://github.com/user-attachments/assets/2770f2da-9b9c-47f4-b6af-092b126d31ea)
![image](https://github.com/user-attachments/assets/3313a0c9-4276-42bb-91d7-c65d41260512)

```suggestion
                    key: "next_chain_preset",
```